MapboxCoreNavigation/UpcomingRouteAlertInfo.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+import CoreLocation
+import MapboxNavigationNative
+import MapboxDirections
+
+/**
+ `UpcomingRouteAlertInfo` encapsulates information about various incoming events. Common attributes like location, distance to the event, length and other is provided for each POI, while specific meta data is supplied via `alert` property.
+ */
+public struct UpcomingRouteAlertInfo {
```

Comment:
Is “upcoming” the right terminology to use? Is there a situation where the developer might end up referring to a past UpcomingRouteAlert? This is sort of like a “positioned” route alert in that it combines the alert itself with positioning information.

Reply:
`UpcomingRouteAlertInfo` are provided to a user on location updates and the feed contains only not passed events. So for the context of how SDK supplies and uses it, "upcoming" seems correct.

Reply:
I think it’s important for the name of a public type to be intrinsic to the data contained inside of it; to some extent, the name should be independent of the situations in which the type is or isn’t used. It would be awkward if the application keeps its own array of passed “upcoming” alerts. (For a contrived example, consider a counter of how many “upcoming” alerts the user has already passed.)